go-controller/pkg/ovn/pods.go

```diff
@@ -107,6 +108,48 @@ func (oc *Controller) deleteLogicalPort(pod *kapi.Pod) {
 		klog.Errorf(err.Error())
 	}
 
+	for _, podIPNet := range portInfo.ips {
+		// delete src-ip cached route to GR
+		nsInfo, err := oc.waitForNamespaceLocked(pod.Namespace)
+		if err != nil {
+			klog.Errorf(err.Error())
+		}
+		podIP := podIPNet.IP.String()
+		if gwToGr, ok := nsInfo.podExternalRoutes[podIP]; ok {
+			if len(gwToGr) == 0 {
+				delete(nsInfo.podExternalRoutes, pod.Status.PodIP)
+				nsInfo.Unlock()
+				continue
+			}
+			var mask string
+			if utilnet.IsIPv6(net.ParseIP(podIP)) {
```

Comment:
@rcarrillocruz @trozet seems like this block could also be a separate function since it's used in two places.

Reply:
done